From f85bdd8dc35b608d03ea4ac88a8cb3cf387c84d2 Mon Sep 17 00:00:00 2001
From: Vignesh Viswanathan <quic_viswanat@quicinc.com>
Date: Mon, 14 Oct 2024 16:22:34 +0530
Subject: [PATCH] arch: Kconfig: Add Config for IO_COHERENCY support

Add Kconfig for ARM and ARM64 for IO_COHERENCY support required for
IPQ5424.

Change-Id: I07202584f9e9c13bbd8216c8ac9ff6724d79b586
Signed-off-by: Vignesh Viswanathan <quic_viswanat@quicinc.com>
---
 arch/arm/mach-qcom/Kconfig | 6 ++++++
 arch/arm64/Kconfig         | 6 ++++++
 2 files changed, 12 insertions(+)

diff --git a/arch/arm/mach-qcom/Kconfig b/arch/arm/mach-qcom/Kconfig
index 54a2e35b47e5..25ec3aee424a 100644
--- a/arch/arm/mach-qcom/Kconfig
+++ b/arch/arm/mach-qcom/Kconfig
@@ -68,4 +68,10 @@ config SKB_FIXED_SIZE_2K
 	  This is a hint to the NSS driver that the ‘skbuff’ size might
 	  need to be fixed at 2KB, to conserve memory.
 
+config IO_COHERENCY
+	bool "Enable IO Coherency for kernel modules"
+	default n
+	help
+	  Enable/Disable IO Coherency for kernel modules.
+
 endif
diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index df6589720b3b..603160521a50 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -2218,6 +2218,12 @@ config UNWIND_PATCH_PAC_INTO_SCS
 config ARCH_IPQ5424
 	bool "Enable for target IPQ5424"
 
+config IO_COHERENCY
+	bool "Enable IO Coherency for kernel modules"
+	default n
+	help
+	  Enable/Disable IO Coherency for kernel modules.
+
 endmenu # "Kernel Features"
 
 menu "Boot options"
-- 
2.34.1

